app/domain/repository/persist_repository_from_upstream.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+class Repository::PersistRepositoryFromUpstream
+  def self.create_or_update_from_host_data(upstream_raw_host_data)
+    return unless upstream_raw_host_data
+
+    ActiveRecord::Base.transaction do
+      existing_repo = find_repository_from_host_data(upstream_raw_host_data)
+
+      if existing_repo.nil?
+        new_repository = Repository.new(uuid: upstream_raw_host_data.repository_uuid, full_name: upstream_raw_host_data.full_name)
+
+        new_repository.host_type = upstream_raw_host_data.host_type
+
+        new_repository.license = upstream_raw_host_data.formatted_license if upstream_raw_host_data.formatted_license
```

Comment:
🎱 Are the postfix ifs needed?

Reply:
yeah, seems like the ones above and below aren't necessary, if it's a new record